package/src/Slider.tsx

```diff
@@ -0,0 +1,273 @@
+     * Should be a plural word, as singular units will be handled.
+     */
+    accessibilityUnits?: string;
+
+    /**
+     * An array of values that represent the different increments displayed
+     * by the slider. All the values passed into this prop must be strings.
+     * Requires passing a value to `accessibilityUnits` to work correctly.
+     * The number of elements must be the same as `maximumValue`.
+     */
+    accessibilityIncrements?: Array<string>;
+  }>;
+
+const SliderComponent = (
+  props: Props,
+  forwardedRef?: Ref<typeof RCTSliderNativeComponent>,
+) => {
+  const style = StyleSheet.compose(styles.slider, props.style);
+
+  const {
+    onValueChange,
+    onSlidingStart,
+    onSlidingComplete,
+    onAccessibilityAction,
+    ...localProps
+  } = props;
+
+  const onValueChangeEvent = onValueChange
+    ? (event: Event) => {
+        onValueChange(event.nativeEvent.value);
+      }
+    : null;
+
+  const _disabled =
+    typeof props.disabled === 'boolean'
+      ? props.disabled
+      : props.accessibilityState?.disabled === true;
+
+  const _accessibilityState =
+    typeof props.disabled === 'boolean'
+      ? {...props.accessibilityState, disabled: props.disabled}
+      : props.accessibilityState;
+
+  const onSlidingStartEvent = onSlidingStart
+    ? (event: Event) => {
+        onSlidingStart(event.nativeEvent.value);
+      }
+    : null;
+  const onSlidingCompleteEvent = onSlidingComplete
+    ? (event: Event) => {
+        onSlidingComplete(event.nativeEvent.value);
+      }
+    : null;
+  const onAccessibilityActionEvent = onAccessibilityAction
+    ? (event: AccessibilityActionEvent) => {
+        onAccessibilityAction(event);
+      }
+    : null;
+
+  return (
+    <RCTSliderNativeComponent
+      {...localProps}
+      accessibilityState={_accessibilityState}
+      thumbImage={
+        Platform.OS === 'web'
+          ? props.thumbImage
+          : props.thumbImage
+          ? Image.resolveAssetSource(props.thumbImage)
+          : undefined
+      }
+      ref={forwardedRef}
+      style={style}
+      onChange={onValueChangeEvent}
+      onRNCSliderSlidingStart={onSlidingStartEvent}
+      onRNCSliderSlidingComplete={onSlidingCompleteEvent}
+      onRNCSliderValueChange={onValueChangeEvent}
+      enabled={!_disabled}
+      disabled={_disabled}
+      onStartShouldSetResponder={() => true}
+      onResponderTerminationRequest={() => false}
+      onRNCSliderAccessibilityAction={onAccessibilityActionEvent}
+    />
+  );
+};
+
+const SliderWithRef = React.forwardRef(SliderComponent);
+
+SliderWithRef.defaultProps = {
+  value: 0,
+  minimumValue: 0,
+  maximumValue: 1,
+  step: 0,
+  inverted: false,
+  tapToSeek: false,
+};
+
+let styles: any;
+if (Platform.OS === 'ios') {
+  styles = StyleSheet.create({
+    slider: {
+      height: 40,
+    },
+  });
+} else {
+  styles = StyleSheet.create({
+    slider: {},
+  });
+}
```

Comment:
I think this can be done by using `Platform.select` or just a simple `Platform.OS` check without typing this to `any`.